saga/sagaState.go

```diff
@@ -6,30 +6,40 @@ import (
 )
 
 /*
+ * Additional data about tasks to be committed to the log
+ * This is Opaque to SagaState, but useful data to persist for
+ * results or debugging
+ */
+type taskData struct {
```

Comment:
If we combine them, it needs to happen in a later code review, since the taskState has been compressed in a separate pull request. 

Also none of this is exposed as part of the public API so less concerned with how we internally represent task state, It can continue to evolve to match our perf / clarity needs, feel free to add a Jira ticket if this is something you feel strongly about.  More concerned with getting it up and running end to end with a cohesive public API